### PR TITLE
Add SASL-SCRAM-SHA-512 support to Kafka Connect and Kafka Connect S2I

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

Add SASL-SCRAM-SHA-512 support to Kafka Connect and Kafka Connect S2I

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

File: documentation/book/con-kafka-connect-authentication.adoc [hunk 1]
```diff
@@ -2,12 +2,16 @@
...
 // assembly-kafka-connect-tls.adoc
 
-[id='ref-kafka-connect-authentication-{context}']
+[id='con-kafka-connect-authentication{context}']
 = Authentication support in Kafka Connect
 
 Authentication can be configured in the `authentication` property in `KafkaConnect.spec` and `KafkaConnectS2I.spec`.
 The `authentication` property specifies the type of the authentication mechanisms which should be used and additional configuration details depending on the mechanism.
-Currently, the only supported authentication type is TLS client authentication.
+Currently, the supported authentication types are:
```
Comment (dmuntima) [inline documentation/book/con-kafka-connect-authentication.adoc]: Currently, the supported authentication .... > The supported authentication ....

File: documentation/book/con-kafka-connect-authentication.adoc [hunk 2]
```diff
@@ -36,4 +40,30 @@ spec:
       certificate: public.crt
       key: private.key
   # ...
+----
+
+== SCRAM-SHA-512 authentication
+
+To use the authentication using the SCRAM-SHA-512 SASL mechanism, set the `type` property to the value `scram-sha-512`.
+SCRAM-SHA-512 authentication is using username and password to authenticate.
```
Comment (dmuntima) [inline documentation/book/con-kafka-connect-authentication.adoc]: SCRAM-SHA-512 authentication is using username and password to authenticate.
The username has to be specified in the `username` property.
The password has to be specified as a link to a `Secret` containing the password in the `passwordSecret` property. >
SCRAM-SHA-512 uses username and password to authenticate. Specify the username in the `username` property. Specify the password as a link to a `Secret` containing the password in the `passwordSecret` property.

File: documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+= Configuring SCRAM-SHA-512 authentication in Kafka Connect
+
+.Prerequisites
+
+* An {ProductPlatformName} cluster
+* A running Cluster Operator
+
+.Procedure
+
+. Find out the username of the user which should be used for authentication.
```
Comment (dmuntima) [inline documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc]: Find out the username of the user which should be used for authentication. - I think we can move this text to the Prerequisites section and rewrite as "Username of the user which should be used for authentication"?

File: documentation/book/con-kafka-connect-authentication.adoc [hunk 4]
```diff
@@ -36,4 +40,30 @@ spec:
       certificate: public.crt
       key: private.key
   # ...
+----
+
+== SCRAM-SHA-512 authentication
+
+To use the authentication using the SCRAM-SHA-512 SASL mechanism, set the `type` property to the value `scram-sha-512`.
+SCRAM-SHA-512 uses username and password to authenticate.
```
Comment (tombentley) [inline documentation/book/con-kafka-connect-authentication.adoc]: uses a username

File: documentation/book/con-kafka-connect-authentication.adoc [hunk 5]
```diff
@@ -36,4 +40,30 @@ spec:
...
+Specify the username in the username property.
```
Comment (tombentley) [inline documentation/book/con-kafka-connect-authentication.adoc]: "...in the `username` property" (i.e. monospace)

File: documentation/book/con-kafka-connect-authentication.adoc [hunk 6]
```diff
@@ -36,4 +40,30 @@ spec:
...
+Specify the password as a link to a Secret containing the password in the passwordSecret property.
```
Comment (tombentley) [inline documentation/book/con-kafka-connect-authentication.adoc]: Secret should be monospace

File: documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+.Prerequisites
+
+* An {ProductPlatformName} cluster
+* A running Cluster Operator
+* Username of the user which should be used for authentication
+
+.Procedure
+
+. Find out the name of the secret with the password which should be used for authentication and the key under which the password is stored in the secret.
+If such secret does not exist yet, prepare a file with the password and create the secret.
```
Comment (tombentley) [inline documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc]: If such a `Secret`...

File: documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+.Prerequisites
+
+* An {ProductPlatformName} cluster
+* A running Cluster Operator
+* Username of the user which should be used for authentication
+
+.Procedure
+
+. Find out the name of the secret with the password which should be used for authentication and the key under which the password is stored in the secret.
```
Comment (tombentley) [inline documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc]: `Secret`

File: documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+If such secret does not exist yet, prepare a file with the password and create the secret.
++
+ifdef::Kubernetes[]
+On {KubernetesName} this can be done using `kubectl create`:
+[source,shell,subs=+quotes]
+echo -n '1f2d1e2e67df' > _<my-password.txt>_
```
Comment (tombentley) [inline documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc]: `echo -n _<password>_ >` (i.e use a placeholder for the password?

File: documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc (hunk 7 above)
Comment (tombentley) [reply documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc]: This still needs to be fixed.

File: documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc [hunk 10]
```diff
@@ -14,7 +14,7 @@
 .Procedure
 
 . Find out the name of the `Secret` with the password which should be used for authentication and the key under which the password is stored in the `Secret`.
-If such secret does not exist yet, prepare a file with the password and create the secret.
+If such `Secret` does not exist yet, prepare a file with the password and create the `Secret`.
```
Comment (tombentley) [inline documentation/book/proc-configuring-kafka-connect-authentication-scram-sha-512.adoc]: It's not juse typography: it should read "If such **a** `Secret`..."

**Pull request merged**